Assets/Scripts/Managers/BattleQueue.cs

```diff
@@ -92,7 +106,105 @@ void RenderBQ()
+        {
+            if (i == 0)
+            {
+                wrapperArray.DisplayWrapperArray();
+                i = 1;
+            }
+
+
+            Wrapper e = array[0];
+
+            ActionClass action = e.ReturnHighest(); // accounts for half-wrappers too 
+
+            if (action == null) 
+            {
+                throw new Exception("Empty Action Returned.");
+            }
+
+            // %%% could be optimised for logic
+            // removing from the Wrapper based on type
+            if (action.IsPlayedByPlayer()) // debugging 
+            {
+                Debug.Log("A player action of speed x is removed: " + action.Speed);
+            }
+            else
+            {
+                Debug.Log("An enemy action of speed x is removed: " + action.Speed);
+            }
+
+            if (e.HighestSpeed == -1 || (e.PlayerAction == null && e.EnemyAction == null))
+            {
+                if (e.EnemyAction != null || e.PlayerAction != null)
+                {
+                    throw new Exception("Wrapper Invariant not Upheld"); // $$$ what are the implications for the coroutine and the exception? 
+                }
+                array.Remove(e); // wrapper is now worthless
+            }
+            else if (e.PlayerAction != null || e.EnemyAction != null)// the wrapper has a component left 
+            {
+                if (action.IsPlayedByPlayer())
+                {
+                    // insert enemy action 
+                    // wrapperArray.InsertEnemyActionIntoWrappers(e.EnemyAction);
+                    wrapperArray.GetWrappers().Remove(e); 
+                    wrapperArray.InsertEnemyActionIntoWrappers(e.EnemyAction);
+                }
+                else
+                {
+                    // insert player action
+                    // wrapperArray.InsertPlayerActionIntoWrappers(e.PlayerAction);
+                    wrapperArray.GetWrappers().Remove(e); 
+                    wrapperArray.InsertPlayerActionIntoWrappers(e.PlayerAction);
+                }
+            }
+            wrapperArray.SortWrappers();
+
+            // wrapperArray.SortWrappers(); // circumspection
+
+            // the same functionality is maintained; BattleQueue is responsible for the wrappers and dewrapping. 
+            yield return StartCoroutine(CardComparator.Instance.ClashCards(action, action)); // $$$ the coroutine suspends it right?
+
+            RenderBQ(); // !!! need to make a new render method. BUT before that need to rely on GameState for intial trigger of wrappers
+
+            actionQueue.GetList().Remove(action);
+
+
+            // how about this:
+            // use original actionQueue to remove
+
+            // check insertion of both enemies and players
+
+        }
+        if (beganFighting)
+        {
+            CombatManager.Instance.GameState = GameState.SELECTION;
+        }
+
+        // ASTER1 
+        roundStart = true;
     }
```

Comment:
This method is difficult to read due to comments and exceptions, the logic pathways are also a little convoluted. Consider cleaning things up and reducing the amount of white space in the code